keyboards/cannonkeys/satisfaction75/satisfaction75.c

```diff
@@ -55,6 +55,54 @@ backlight_config_t kb_backlight_config = {
 };
 
 #ifdef VIA_ENABLE
+
+void backlight_get_value( uint8_t *data )
+{
+	uint8_t *value_id = &(data[0]);
+	uint8_t *value_data = &(data[1]);
+  switch (*value_id)
+  {
+    case id_qmk_backlight_brightness:
+    {
+      // level / BACKLIGHT_LEVELS * 255
+      value_data[0] = ((uint16_t)kb_backlight_config.level) * 255 / BACKLIGHT_LEVELS;
+      break;
+    }
+    case id_qmk_backlight_effect:
+    {
+      value_data[0] = kb_backlight_config.breathing ? 1 : 0;
+      break;
+    }
+  }
+}
+
+void backlight_set_value( uint8_t *data )
+{
+	uint8_t *value_id = &(data[0]);
+	uint8_t *value_data = &(data[1]);
+  switch (*value_id)
+  {
+    case id_qmk_backlight_brightness:
+    {
+      // level / 255 * BACKLIGHT_LEVELS
+      kb_backlight_config.level = ((uint16_t)value_data[0]) * BACKLIGHT_LEVELS / 255;
+      backlight_set(kb_backlight_config.level);
+      break;
+    }
+    case id_qmk_backlight_effect:
+    {
+      if ( value_data[0] == 0 ) {
+        kb_backlight_config.breathing = false;
+        breathing_disable();
+      } else {
+        kb_backlight_config.breathing = true;
+        breathing_enable();
+      }
+      break;
+    }
+  }
+}
+
 void raw_hid_receive_kb( uint8_t *data, uint8_t length )
```

Comment:
`length` seems to be an unused parameter in this routine.

Reply:
@mechmerlin an artefact of QMK's style? (Passing down all variables from a quantum level function to a kb level function)

Reply:
The core raw HID functions have this length parameter, because you kinda need it when you're passed a pointer - you have to know when to stop reading ;) Keyboard- and user/keymap-level callbacks should have the same signature, so this is fine.